src/components/Table/columnSizes.ts

```diff
@@ -0,0 +1,73 @@
+import { useResizeObserver } from "@react-aria/utils";
+import { MutableRefObject, useCallback, useRef, useState } from "react";
+import { calcColumnSizes, GridColumn, GridStyle } from "src/components/Table/GridTable";
+import { useDebouncedCallback } from "use-debounce";
+
+/**
+ * Calculates an array of sizes for each of our columns.
+ *
+ * We originally supported CSS grid-template-column definitions which allowed fancier,
+ * dynamic/content-based widths, but have eventually dropped it mainly due to:
+ *
+ * 1. In virtual tables, a) the table never has all of the rows in DOM at a single time,
+ * so any "content-based" widths will change as you scroll the table, which is weird, and
+ * b) a sticky header and rows are put in different DOM parent elements by react-virtuoso,
+ * so wouldn't arrive at the same "content-based" widths.
+ *
+ * 2. Using CSS grid but still have a row-level div for hover/focus targeting required
+ * a "fake" `display: contents` div that couldn't have actually any styles applied to it.
+ *
+ * So we've just got with essentially fixed/deterministic widths, i.e. `px` or `percent` or
+ * `fr`.
+ *
+ * Disclaimer that we roll our own `fr` b/c we're not in CSS grid anymore.
+ *
+ * Also, disclaimer that the capital-S `SetupColumnSizes` name is terrible, but it placates
+ * the eslint rule about "hooks can only be used in components", which has good intentions,
+ * but because this is deterministically called from GridTable, we still follow the rules
+ * of hooks.
```

Comment:
Why not just call it useSetupColumnSize()? Looks like a custom hook. Smells like a custom hook. Just call it a custom hook 🤔

Reply:
Lol, b/c I forgot about that. Yay code previews :-D

Reply:
@TylerR909 fixed!